modules/frontend/querysharding.go

```diff
@@ -194,6 +194,16 @@ func mergeResponses(ctx context.Context, marshallingFormat string, rrs []Request
 		}, nil
 	}
 
+	traceObject := &tempopb.Trace{}
+	err := proto.Unmarshal(combinedTrace, traceObject)
+	if err != nil {
+		return nil, err
+	}
+	traceObject, err = DedupeSpanIDs(ctx, traceObject)
```

Comment:
should this be injected as a new middleware pipeline thing instead of snuck into the end of querysharding?

Reply:
Yep was a little lazy hehe but you're right, going to implement as a middleware.